src/AlarmClock.h

```diff
@@ -39,11 +50,11 @@ template<typename Duration> class AlarmClock {
    }
 
    void Reset() {
-      if (!mExpired.load()) {
-         StopBackgroundThread();
-      }
-      mExpired.store(false); 
-      mExited = std::async(std::launch::async, &AlarmClock::AlarmClockThread, this);
+      unique_lock<mutex> lck(mMutex);
+      mReset.store(true);
+      mExpired.store(0);
```

Comment:
How would that assertion work? I've never done that in cpp. We can pair tomorrow if it's too much to explain via comment.